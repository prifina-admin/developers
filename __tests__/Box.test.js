import React, {createRef} from "react";
import renderer from "react-test-renderer";
import Box from "../src/Box";

test("Box renders correctly", () => {
    const tree = renderer.create( <Box
        width={[1 / 2]}
        height={"100px"}
        border={"1px solid"}
        borderRadius={4}
        p={2}
    >
        Something
    </Box> ).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Box renders sth with Ref correctly", () => {
    const boxRef = createRef();
    const tree2 = renderer.create( <Box
        width={[1 / 2]}
        height={"100px"}
        border={"1px solid"}
        borderRadius={4}
        p={2}
        ref={boxRef}
        foo={"bar"}
    >
        Something with Ref
    </Box> ).toJSON();
    expect(tree2).toMatchSnapshot();
});