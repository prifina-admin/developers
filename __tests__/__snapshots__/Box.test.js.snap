// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
<div
  className="Box__BoxElement-sc-1iw4lg9-0 jsloqb"
>
  Something
</div>
`;

exports[`Box renders sth with Ref correctly 1`] = `
<div
  className="Box__BoxElement-sc-1iw4lg9-0 jsloqb"
  foo="bar"
>
  Something with Ref
</div>
`;
